File: problem_solving/8_MiniMaxSum.rb
def miniMaxSum(arr)
  sum = arr.sum
  min = sum - arr.max
  max = sum - arr.min
  puts "#{min} #{max}"
end

arr = [*(1..5)]
miniMaxSum(arr)

# # ====================
# arrの中の値の一番大きな値を除いた合計(min)と
# 一番小さな値を除いた合計(max)を求める問題。

# arr中の全ての値の合計を出し、
# 合計値からarr中の最小値を引いて得られた値を求める
# 合計値からarr中の最大値を引いて得られた値を求める

# # ====================
# get sum of all values in the arr
# subtract the smallest value of the arr from the sum
# subtract the biggest value in the arr from the sum

File: problem_solving/8_MiniMaxSum.js
function miniMaxSum(arr) {
  let total = 0;
  let max = arr.reduce((prev, current) => prev + current, total) - Math.min(...arr)
  let min = arr.reduce((prev, current) => prev + current, total) - Math.max(...arr)
  console.log(`${min} ${max}`)
}

arr = [1, 2, 3, 4, 5]
miniMaxSum(arr)

// ====================
// 問題:
//   arrの中の値の一番大きな値を除いた合計(min)と
//   一番小さな値を除いた合計(max)を求める問題。

// ポイント解説:
//   arr.reduce((prev, current) => prev + current, total)
//     arr中の全ての値の合計を求める。

//   Math.min(...arr)
//     arr中の最小値を取得。
//     ... スプレッド構文は配列を展開する。

//   Math.max(...arr)
//     arr中の最大値を取得。

// ====================
// arr.reduce((prev, current) => prev + current, total)
//   Total value of arr.

// Math.min(...arr)
//   the smallest value in the arr
//   ... User spread syntax to expand array

// Math.max(...arr)
//   the greatest value in the arr